# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsRefreshDriver.cpp
# Commit: 3508e2bd4e3c
# Full Hash: 3508e2bd4e3cbc756cff7da7aee3c60c29c15bd1
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2024-02-29 21:06:42
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1880928 - Remove last remembered size observer, handle it in Document::DetermineProximityToViewportAndNotifyResizeObservers. r=emilio,layout-reviewers
#   
#   The CSS Box Sizing specification indicates that last remembered sizes
#   are recorded "at the time that ResizeObserver events are determined and
#   delivered" [1].
# ==============================================================================

diff -r 467f563c9ea5 -r 3508e2bd4e3c layout/base/nsRefreshDriver.cpp
--- a/layout/base/nsRefreshDriver.cpp	Thu Feb 29 08:50:55 2024 +0000
+++ b/layout/base/nsRefreshDriver.cpp	Thu Feb 29 08:50:55 2024 +0000
@@ -2264,7 +2264,8 @@
       return false;
     }
     return ps->HasContentVisibilityAutoFrames() ||
-           aDocument->HasResizeObservers();
+           aDocument->HasResizeObservers() ||
+           aDocument->HasElementsWithLastRememberedSize();
   };
 
   AutoTArray<RefPtr<Document>, 32> documents;
